backend/src/main/scala/cromwell/backend/validation/ValidatedRuntimeAttributesBuilder.scala

```diff
@@ -54,32 +53,26 @@ trait ValidatedRuntimeAttributesBuilder {
 
     val runtimeAttributesErrorOr: ErrorOr[ValidatedRuntimeAttributes] = validate(attrs)
     runtimeAttributesErrorOr match {
-      case Success(runtimeAttributes) => runtimeAttributes
-      case Failure(nel) => throw new RuntimeException with MessageAggregation {
+      case Valid(runtimeAttributes) => runtimeAttributes
+      case Invalid(nel) => throw new RuntimeException with MessageAggregation {
         override def exceptionContext: String = "Runtime attribute validation failed"
 
-        override def errorMessages: Traversable[String] = nel.list.toList
+        override def errorMessages: Traversable[String] = nel.toList
       }
     }
   }
 
   private def validate(values: Map[String, WdlValue]): ErrorOr[ValidatedRuntimeAttributes] = {
-    val validationsForValues: Seq[RuntimeAttributesValidation[_]] = validations ++ unsupportedExtraValidations
-    val errorsOrValuesMap: Seq[(String, ErrorOr[Any])] =
-      validationsForValues.map(validation => validation.key -> validation.validate(values))
-
-    import scalaz.Scalaz._
+    val validationsForValues = validations ++ unsupportedExtraValidations
+    val listOfKeysToErrorOrAnys: List[(String, ErrorOr[Any])] =
+      validationsForValues.map(validation => validation.key -> validation.validate(values)).toList
 
-    val emptyResult: ErrorOr[List[(String, Any)]] = List.empty[(String, Any)].success
-    val validationResult = errorsOrValuesMap.foldLeft(emptyResult) { (agg, errorOrValue) =>
-      agg +++ {
-        errorOrValue match {
-          case (key, Success(value)) => List(key -> value).success
-          case (key, Failure(nel)) => nel.failure
-        }
-      }
+    val listOfErrorOrKeysToAnys: List[ErrorOr[(String, Any)]] = listOfKeysToErrorOrAnys map {
+      case (key, errorOrAny) => errorOrAny map { any => (key, any) }
     }
 
-    validationResult.map(result => ValidatedRuntimeAttributes(result.toMap))
+    import cats.syntax.traverse._
+    val errorOrListOfKeysToAnys: ErrorOr[List[(String, Any)]] = listOfErrorOrKeysToAnys.sequence[ErrorOr, (String, Any)]
+    errorOrListOfKeysToAnys map { listOfKeysToAnys => ValidatedRuntimeAttributes(listOfKeysToAnys.toMap) }
```

Comment:
@kshakir I looked at this for a while and basically I don't think the types fit for a `Monoid#combineAll`.  While there is a `Monoid` instance for `Map`, it requires the `Map` values to have `Semigroup` instances.  This is because a `Map` semigroup/monoid would have to be able to combine values associatively if two entries have the same key.  But the values here are just `Any`s and for entries with the same key the `toMap` logic throws away one or the other value arbitrarily.

I did manage to clean this up a bit with some `sequence`ing though the variable names are now painfully explicit.  😄